packages/flutter/lib/src/services/message_codecs.dart

```diff
@@ -386,9 +386,28 @@ class StandardMessageCodec implements MessageCodec<Object?> {
       }
     } else if (value is String) {
       buffer.putUint8(_valueString);
-      final Uint8List bytes = utf8.encoder.convert(value);
-      writeSize(buffer, bytes.length);
-      buffer.putUint8List(bytes);
+      final Uint8List asciiBytes = Uint8List(value.length);
+      Uint8List? utf8Bytes;
+      int utf8Offset = 0;
+      // Only do utf8 encoding if we encounter non-ascii characters.
+      for(int i = 0; i < value.length; ++i) {
+        final int char = value.codeUnitAt(i);
+        if (char <= 0x7f) {
+          asciiBytes[i] = char;
+        } else {
+          utf8Bytes = utf8.encoder.convert(value.substring(i));
+          utf8Offset = i;
+          break;
+        }
+      }
+      if (utf8Bytes != null) {
+        writeSize(buffer, utf8Offset + utf8Bytes.length);
+        buffer.putUint8List(asciiBytes.sublist(0, utf8Offset));
```

Comment:
I think you could avoid a sublist here by constructing a view

Reply:
I wondered about that.  I can try measuring it quickly.  I'll add a utf8 performance benchmark.

Reply:
Done, added a benchmark, it went from 0.86337 µs to 0.75239 µs